### PR TITLE
Issue/3206 test module update command

### LINKED ISSUE #3206: modules v2 pypi loader test: test moduletool update command
Add test for `inmanta module update`. Think about any edge cases.

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

Added tests for the `inmanta module update` command.

closes #3206 

# Self Check:

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: tests/moduletool/test_update.py [hunk 1]
```diff
@@ -66,3 +77,87 @@ def test_module_update_with_install_mode_master(
...
+
+    pip_index = PipIndex(artifact_dir=os.path.join(str(tmpdir), "pip-index"))
+
+    def assert_version_installed(module_name: str, version: str) -> None:
+        package_name = ModuleV2.get_package_name_for(module_name)
+        installed_packages: Dict[str, Version] = process_env.get_installed_packages()
+        assert package_name in installed_packages
+        assert str(installed_packages[package_name]) == version
+
+    for module_name, versions in {"module2": ["2.0.1", "2.1.0", "2.2.0", "3.0.1"], "module1": ["1.2.4", "1.2.5"]}.items():
```
Comment (wouterdb) [inline tests/moduletool/test_update.py:102]: This is really cool!

I would document why the module structure is set up like this specifically, because I have the feeling that some subtly eludes me.

**Comment by arnaudsjs** [comment]
I noticed I forgot to take into account the `install_mode` set on the project during the implementation of the `inmanta module install` command. I am adding it now.

File: src/inmanta/env.py [hunk 2]
```diff
@@ -104,7 +104,11 @@ def get_installed_packages(self, only_editable: bool = False) -> Dict[str, versi
         return {r["name"]: version.Version(r["version"]) for r in json.loads(output)}
 
     def install_from_index(
-        self, requirements: List[Requirement], index_urls: Optional[List[str]] = None, upgrade: bool = False
+        self,
+        requirements: List[Requirement],
+        index_urls: Optional[List[str]] = None,
+        upgrade: bool = False,
+        allow_pre_releases: bool = False,
```
Comment (wouterdb) [inline src/inmanta/env.py:111]: Does inmanta projects install use this flag correctly?  (or is that put in the relevant ticket?)

**Review by sanderr — APPROVED** [review]

File: src/inmanta/env.py (hunk 2 above)
Comment (arnaudsjs) [reply src/inmanta/env.py:111]: I don't think it does and there doesn't seem to be a ticket for that. I was planning to create a ticket, but maybe it's easier to include it in this PR. I will make the change.

I will also add some user documentation on how the `install_mode` option is interpreted for V2 modules. That's missing as well.

**Review by sanderr — APPROVED** [review]
The module template change for dev versions is a definite improvement!

**Review by wouterdb — APPROVED** [review]
